packages/contracts-bedrock/test/setup/ForkLive.s.sol

```diff
@@ -49,20 +49,33 @@ contract ForkLive is Deployer {
 
     /// @notice Forks, upgrades and tests a production network.
     /// @dev This function sets up the system to test by:
-    ///      1. reading the superchain-registry to get the contract addresses we wish to test from that network.
-    ///      2. deploying the updated OPCM and implementations of the contracts.
-    ///      3. upgrading the system using the OPCM.upgrade() function.
+    ///      0. reading the environment variable to determine the base and OP chain names, read the STATE_PATH data
+    /// created from superchain ops.
+    ///      1. if the environment variable is set, load the state from the given path.
+    ///      2. read the superchain-registry to get the contract addresses we wish to test from that network.
+    ///      3. deploying the updated OPCM and implementations of the contracts.
+    ///      4. upgrading the system using the OPCM.upgrade() function if not using the applied state from superchain
+    /// ops.
     function run() public {
-        // Read the superchain registry and save the addresses to the Artifacts contract.
-        _readSuperchainRegistry();
-
-        // Now deploy the updated OPCM and implementations of the contracts
-        _deployNewImplementations();
+        bool useOpsRepo = bytes(vm.envOr("STATE_PATH", string(""))).length > 0;
+        if (useOpsRepo) {
+            console.log("ForkLive: loading state from %s", vm.envString("STATE_PATH"));
```

Comment:
I don't think we need to read the environment variable twice. We could read it once at the top of the run function. If it's empty then follow the traditional route, if not then follow new, superchain-ops route.